--- src/main/kotlin/club/sk1er/mods/levelhead/commands/LevelheadCommand.kt
+++ src/main/kotlin/club/sk1er/mods/levelhead/commands/LevelheadCommand.kt
@@ -1050,11 +1050,12 @@
         msg2.appendSibling(ChatComponentText("${ChatColor.YELLOW}, show self "))

         val toggleText = if (showSelf) "on" else "off"
+        val oppositeToggleText = if (showSelf) "off" else "on"
         val toggleColor = if (showSelf) ChatColor.GREEN else ChatColor.RED
         msg2.appendSibling(ChatComponentText("$toggleColor$toggleText").apply {
-            chatStyle.chatClickEvent = ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, "/levelhead display showself $toggleText")
-            chatStyle.chatHoverEvent = HoverEvent(HoverEvent.Action.SHOW_TEXT, ChatComponentText("${ChatColor.GREEN}Click to edit show self"))
+            chatStyle.chatClickEvent = ClickEvent(ClickEvent.Action.RUN_COMMAND, "/levelhead display showself $oppositeToggleText")
+            chatStyle.chatHoverEvent = HoverEvent(HoverEvent.Action.SHOW_TEXT, ChatComponentText("${ChatColor.GREEN}Click to toggle show self"))
         })
         msg2.appendSibling(ChatComponentText("${ChatColor.YELLOW}."))
         sendMessage(msg2)
